File: BeenHouse/app/src/main/java/com/gbabtech/beenhouse/activities/fragments/MenuFragment.java
package com.gbabtech.beenhouse.activities.fragments;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import com.gbabtech.beenhouse.R;
import com.gbabtech.beenhouse.activities.MainMenuAdapter;
import butterknife.ButterKnife;

public class MenuFragment extends Fragment {

    public MenuFragment() {
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_menu, container, false);
        ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        View currentView = getView();
        assert currentView != null;
        RecyclerView menuRecyclerView = currentView.findViewById(R.id.menuRecyclerView);
        initRecyclerView(menuRecyclerView);
    }

    private void initRecyclerView(RecyclerView menuRecyclerView) {
        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(getActivity());
        ((LinearLayoutManager) layoutManager).setOrientation(LinearLayoutManager.VERTICAL);
        menuRecyclerView.setLayoutManager(layoutManager);

        MainMenuAdapter adapter = new MainMenuAdapter();
        menuRecyclerView.setAdapter(adapter);
    }
}

File: bookit.ng/app/src/main/java/com/bookitng/bookitng/fragments/MenuFragment.kt
package com.bookitng.bookitng.fragments

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bookitng.bookitng.R
import com.bookitng.bookitng.activities.MainMenuActivity
import com.bookitng.bookitng.utils.Constants
import kotlinx.android.synthetic.main.fragment_menu.*


class MenuFragment : Fragment(), View.OnClickListener {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        (activity as MainMenuActivity).getBottomNav().visibility = View.GONE
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_menu, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        flightsLayout.setOnClickListener(this)
        hotelsLayout.setOnClickListener(this)
        vacationsLayout.setOnClickListener(this)
        busesLayout.setOnClickListener(this)
        entertainmentLayout.setOnClickListener(this)

    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.flightsLayout -> (activity as MainMenuActivity).getBottomNav()
                .findViewById<View>(R.id.navigation_flights).performClick()

            R.id.hotelsLayout -> (activity as MainMenuActivity).getBottomNav()
                .findViewById<View>(R.id.navigation_hotels).performClick()

            R.id.vacationsLayout -> (activity as MainMenuActivity).getBottomNav()
                .findViewById<View>(R.id.navigation_vacations).performClick()

            R.id.busesLayout -> (activity as MainMenuActivity).getBottomNav()
                .findViewById<View>(R.id.navigation_buses).performClick()

            R.id.entertainmentLayout -> (activity as MainMenuActivity).getBottomNav()
                .findViewById<View>(R.id.navigation_entertainment).performClick()
        }
    }


}
